plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/delete/TestBase85Codec.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.delete;
+
+import org.testng.annotations.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.deltalake.delete.Base85Codec.BASE;
+import static io.trino.plugin.deltalake.delete.Base85Codec.BASE_2ND_POWER;
+import static io.trino.plugin.deltalake.delete.Base85Codec.BASE_3RD_POWER;
+import static io.trino.plugin.deltalake.delete.Base85Codec.BASE_4TH_POWER;
+import static io.trino.plugin.deltalake.delete.Base85Codec.ENCODE_MAP;
+import static io.trino.plugin.deltalake.delete.Base85Codec.decode;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestBase85Codec
+{
+    @Test
+    public void testDecodeBlocksIllegalCharacter()
+    {
+        assertThatThrownBy(() -> decode("ab" + 0x7F + "de")).hasMessageContaining("Input should be 5 character aligned");
+
+        assertThatThrownBy(() -> decode("abîde")).hasMessageContaining("Input character is not ASCII: [î]");
+        assertThatThrownBy(() -> decode("abπde")).hasMessageContaining("Input character is not ASCII: [π]");
+        assertThatThrownBy(() -> decode("ab\"de")).hasMessageContaining("Invalid input character: [\"]");
+    }
+
+    @Test
+    public void testEncodeBytes()
+    {
+        // The test case comes from https://rfc.zeromq.org/spec/32
+        byte[] inputBytes = new byte[] {(byte) 0x86, 0x4F, (byte) 0xD2, 0x6F, (byte) 0xB5, 0x59, (byte) 0xF7, 0x5B};
+        String encoded = encodeBytes(inputBytes);
+        assertThat(encoded).isEqualTo("HelloWorld");
+    }
+
+    @Test
+    public void testCodecRoundTrip()
+    {
+        assertThat(encodeBytes(decodeBytes("HelloWorld", 8)))
+                .isEqualTo("HelloWorld");
+        assertThat(encodeBytes(decodeBytes("wi5b=000010000siXQKl0rr91000f55c8Xg0@@D72lkbi5=-{L", 40)))
+                .isEqualTo("wi5b=000010000siXQKl0rr91000f55c8Xg0@@D72lkbi5=-{L");
+    }
+
+    @Test
+    public void testDecodeBytes()
+    {
+        String data = "HelloWorld";
+        byte[] bytes = decodeBytes(data, 8);
+        assertThat(bytes).isEqualTo(new byte[] {(byte) 0x86, 0x4F, (byte) 0xD2, 0x6F, (byte) 0xB5, 0x59, (byte) 0xF7, 0x5B});
```

Comment:
Suppress `Cast from 'int' to 'byte' may result in loss of precision` Intellij's warning